In [2]:
from google.colab import files
files.upload()

Saving kaggle.json to kaggle.json


{'kaggle.json': b'{"username":"hydrohelium","key":"ea6a3fcd9ef2c15fde801a0d8fedaf6b"}'}

In [3]:
!rm -r ~/.kaggle
!mkdir ~/.kaggle
!mv ./kaggle.json ~/.kaggle/
!chmod 600 ~/.kaggle/kaggle.json

rm: cannot remove '/root/.kaggle': No such file or directory


In [4]:
!kaggle competitions download -c titanic

  0% 0.00/34.1k [00:00<?, ?B/s]
100% 34.1k/34.1k [00:00<00:00, 61.5MB/s]


In [5]:
!mkdir Dataset
!cp /content/titanic.zip /content/Dataset/titanic.zip
!unzip -q /content/Dataset/titanic.zip -d /content/Dataset
!rm /content/Dataset/titanic.zip

In [6]:
!ls /content/Dataset/

gender_submission.csv  test.csv  train.csv


In [7]:
import pandas as pd
data = pd.read_csv("/content/Dataset/train.csv")
data.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

In [8]:
## Feature selection:- 

## Test A:- Gender, Pcalss, Age, Parch, Fare, Embarked 
refined_data = data[["Survived", "Pclass","Sex", "Age","Fare", "Embarked"]]
print(refined_data)
print(refined_data.dtypes)

     Survived  Pclass     Sex   Age     Fare Embarked
0           0       3    male  22.0   7.2500        S
1           1       1  female  38.0  71.2833        C
2           1       3  female  26.0   7.9250        S
3           1       1  female  35.0  53.1000        S
4           0       3    male  35.0   8.0500        S
..        ...     ...     ...   ...      ...      ...
886         0       2    male  27.0  13.0000        S
887         1       1  female  19.0  30.0000        S
888         0       3  female   NaN  23.4500        S
889         1       1    male  26.0  30.0000        C
890         0       3    male  32.0   7.7500        Q

[891 rows x 6 columns]
Survived      int64
Pclass        int64
Sex          object
Age         float64
Fare        float64
Embarked     object
dtype: object


In [9]:
m_surviours = refined_data.loc[(refined_data['Survived'] == 1) & (refined_data['Sex']=="male")]
f_survivours = refined_data.loc[(refined_data['Survived'] == 1) & (refined_data['Sex']=="female")]

print(m_surviours.shape[0])
print(f_survivours.shape[0])

109
233


In [10]:
m_surviours.head(10)
temp1 = m_surviours.dropna(subset=['Age'])
print(temp1.head(10))
print(temp1.describe())

     Survived  Pclass   Sex    Age     Fare Embarked
21          1       2  male  34.00  13.0000        S
23          1       1  male  28.00  35.5000        S
74          1       3  male  32.00  56.4958        S
78          1       2  male   0.83  29.0000        S
81          1       3  male  29.00   9.5000        S
97          1       1  male  23.00  63.3583        C
125         1       3  male  12.00  11.2417        C
127         1       3  male  24.00   7.1417        S
146         1       3  male  27.00   7.7958        S
165         1       3  male   9.00  20.5250        S
       Survived     Pclass        Age        Fare
count      93.0  93.000000  93.000000   93.000000
mean        1.0   1.978495  27.276022   43.793865
std         0.0   0.920521  16.504803   76.649011
min         1.0   1.000000   0.420000    0.000000
25%         1.0   1.000000  18.000000    9.500000
50%         1.0   2.000000  28.000000   26.287500
75%         1.0   3.000000  36.000000   52.554200
max         1.0  

In [11]:
temp2 = f_survivours.dropna(subset=['Age'])
print(temp2.head(10))
print(temp2.describe())

    Survived  Pclass     Sex   Age     Fare Embarked
1          1       1  female  38.0  71.2833        C
2          1       3  female  26.0   7.9250        S
3          1       1  female  35.0  53.1000        S
8          1       3  female  27.0  11.1333        S
9          1       2  female  14.0  30.0708        C
10         1       3  female   4.0  16.7000        S
11         1       1  female  58.0  26.5500        S
15         1       2  female  55.0  16.0000        S
22         1       3  female  15.0   8.0292        Q
25         1       3  female  38.0  31.3875        S
       Survived      Pclass         Age        Fare
count     197.0  197.000000  197.000000  197.000000
mean        1.0    1.822335   28.847716   55.643148
std         0.0    0.791478   14.175073   67.209168
min         1.0    1.000000    0.750000    7.225000
25%         1.0    1.000000   19.000000   13.000000
50%         1.0    2.000000   28.000000   26.250000
75%         1.0    2.000000   38.000000   77.958300
m

## Mean age among male surviours is 27.27 
## Mean age among female surviours is 26.25 


In [15]:
m_died = refined_data.loc[(refined_data['Survived'] == 0) & (refined_data['Sex']=="male")]
f_died = refined_data.loc[(refined_data['Survived'] == 0) & (refined_data['Sex']=="female")]

print(m_died.shape[0])
print(f_died.shape[0])

468
81


In [17]:
print(m_died.describe())
print(f_died.describe())

       Survived      Pclass         Age        Fare
count     468.0  468.000000  360.000000  468.000000
mean        0.0    2.476496   31.618056   21.960993
std         0.0    0.761411   14.056019   32.409799
min         0.0    1.000000    1.000000    0.000000
25%         0.0    2.000000   21.750000    7.795800
50%         0.0    3.000000   29.000000    9.416650
75%         0.0    3.000000   39.250000   26.000000
max         0.0    3.000000   74.000000  263.000000
       Survived     Pclass        Age        Fare
count      81.0  81.000000  64.000000   81.000000
mean        0.0   2.851852  25.046875   23.024385
std         0.0   0.450309  13.618591   24.821287
min         0.0   1.000000   2.000000    6.750000
25%         0.0   3.000000  16.750000    8.850000
50%         0.0   3.000000  24.500000   15.245800
75%         0.0   3.000000  33.250000   27.900000
max         0.0   3.000000  57.000000  151.550000


### Replacing the age with mean age and making a DNN for prediction
## Convention 
female = [0, 1]  
male = [1, 0 ] 

S, C, Q -> 1,2,3  

In [12]:
no_nadata= refined_data.dropna(subset=['Age'])
no_nadata.loc[(no_nadata.Sex)=='male', 'Sex']=0
no_nadata.loc[(no_nadata.Sex)=='female', 'Sex']=1
print(no_nadata.dtypes)
no_nadata['Sex'] = no_nadata['Sex'].astype(str).astype(int)
print(no_nadata.dtypes)

Survived      int64
Pclass        int64
Sex          object
Age         float64
Fare        float64
Embarked     object
dtype: object
Survived      int64
Pclass        int64
Sex           int64
Age         float64
Fare        float64
Embarked     object
dtype: object


<ipython-input-12-3725719b9eb9>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  no_nadata['Sex'] = no_nadata['Sex'].astype(str).astype(int)


In [16]:
no_nadata= refined_data.dropna(subset=['Age'])
no_nadata["0"]=[0 if i=="male" else 1 for i in no_nadata["Sex"].values]
no_nadata["1"]=[1 if i=="male" else 0 for i in no_nadata["Sex"].values]
## [S,C, ]
no_nadata["2"]=[]


no_nadata.head(5)

<ipython-input-16-2016c670867a>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  no_nadata["0"]=[0 if i=="male" else 1 for i in no_nadata["Sex"].values]
<ipython-input-16-2016c670867a>:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  no_nadata["1"]=[1 if i=="male" else 0 for i in no_nadata["Sex"].values]


Survived  Pclass     Sex   Age     Fare Embarked  0  1
0         0       3    male  22.0   7.2500        S  0  1
1         1       1  female  38.0  71.2833        C  1  0
2         1       3  female  26.0   7.9250        S  1  0
3         1       1  female  35.0  53.1000        S  1  0
4         0       3    male  35.0   8.0500        S  0  1

In [34]:
from sklearn.preprocessing import OneHotEncoder
encoder = OneHotEncoder(handle_unknown='ignore')
encoder_df = pd.DataFrame(encoder.fit_transform(no_nadata[['Sex','Embarked','Pclass']]).toarray())
final_df = no_nadata.join(encoder_df)
final_df.drop('Sex', axis=1, inplace=True)
final_df.drop('Embarked', axis=1, inplace=True)
final_df.drop('Pclass', axis=1, inplace=True)
final_df.head(10)

Survived   Age     Fare    0    1    2    3    4    5    6    7    8
0          0  22.0   7.2500  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  1.0
1          1  38.0  71.2833  0.0  1.0  1.0  0.0  0.0  0.0  1.0  0.0  0.0
2          1  26.0   7.9250  0.0  1.0  0.0  0.0  1.0  0.0  0.0  0.0  1.0
3          1  35.0  53.1000  0.0  1.0  0.0  0.0  1.0  0.0  1.0  0.0  0.0
4          0  35.0   8.0500  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  1.0
6          0  54.0  51.8625  1.0  0.0  0.0  0.0  1.0  0.0  0.0  0.0  1.0
7          0   2.0  21.0750  0.0  1.0  0.0  0.0  1.0  0.0  0.0  0.0  1.0
8          1  27.0  11.1333  0.0  1.0  1.0  0.0  0.0  0.0  0.0  1.0  0.0
9          1  14.0  30.0708  0.0  1.0  0.0  0.0  1.0  0.0  0.0  0.0  1.0
10         1   4.0  16.7000  0.0  1.0  0.0  0.0  1.0  0.0  1.0  0.0  0.0

In [ ]:
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers

In [ ]:
model = keras.Sequential([
    layers.Dense()
])